llvm/lib/Target/EVM/EVMTargetMachine.cpp

```diff
@@ -212,6 +228,17 @@ class EVMPassConfig final : public TargetPassConfig {
 void EVMPassConfig::addIRPasses() {
   addPass(createEVMLowerIntrinsicsPass());
   if (TM->getOptLevel() != CodeGenOptLevel::None) {
+    // Call EarlyCSE pass to find and remove subexpressions in the lowered
+    // result.
+    addPass(createEarlyCSEPass(true));
+    addPass(createGVNPass());
+    // The DSE pass we run at the end of the optimization pipeline may open
```

Comment:
`why`s or examples are welcome. Otherwise, I suggest rephrasing it as we _noticed_ that sometimes DSE makes it possible to simplify CFG, i.e. its the matter of observation - not a mere belief.